commit 90e102335bc45af76455a8591ca3b2d5b0aa6e44
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Mon Feb 20 20:05:26 2017 +0800

    Only return QPixmap when it is not null
    
    QFile::exists() returns true even for executables in $PATH, and loading
    those files with QPixmap() will return a null object.
    
    Change-Id: I29a954aa2d504ca051b7328978426b6b993da753

diff --git a/frame/util/themeappicon.cpp b/frame/util/themeappicon.cpp
index b8e1587..b150b97 100644
--- a/frame/util/themeappicon.cpp
+++ b/frame/util/themeappicon.cpp
@@ -18,8 +18,11 @@ QPixmap ThemeAppIcon::getIcon(const QString iconName, const int size)
 {
     QPixmap pixmap(size, size);
 
-    if (QFile::exists(iconName))
-        return QPixmap(iconName);
+    if (QFile::exists(iconName)) {
+        pixmap = QPixmap(iconName);
+        if (!pixmap.isNull())
+            return pixmap;
+    }
     if (iconName.startsWith("data:image/"))
     {
         const QStringList strs = iconName.split("base64,");
